frontend/HelperFuncs/utils.js

```diff
@@ -219,4 +219,31 @@ export function isValidPassword(password){
         return result
     }
     return result
+}
+
+
+/*
+helper function to convert the dictionary form of a search object selected by a user into a 
+readable URL endpoint for navigaion. Each category adheres to the following unique form factors: 
+city: 'Menlo Park, CA' -> 'menloparkCA'
+salary: '$87,076 - $170,050' -> '87076-170050'
+job: 'Software Enigneer' -> 'SoftwareEngineer'
+requires [search] to be a dictionary with fields label and category 
+*/
+export function searchRouteFormatter(search) {
```

Comment:
how will we handle unexpected characters?

Reply:
This is a good point, although job title characters will in theory never be truly unexpected since the number of job titles is finite thanks to the use of the API (controlling for these kinds of scenarios is exactly why I though implementing the API was worthwhile). For example, one of the more edge-case looking job titles in the database 'An/Ssn-2 (V) 4 Operator' gets routed to the page `http://localhost:5173/search/An/Ssn-2(V)4Operator` which is an issue because the additional `/` in the job title might cause some trouble. I'll make a commit to adjust the function to use `encodeURIComponen` which should cover this and allow for better spacing than camel case.

Reply:
<img width="601" alt="Screenshot 2024-07-12 at 10 42 03 AM" src="https://github.com/user-attachments/assets/2e48bd3b-d3c5-4018-9a5d-66d37a2807ef">


this should now be fixed. in 96fee21